core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -653,8 +660,9 @@ private Expression translate0(RexNode expr, RexImpTable.NullAs nullAs,
     case CORREL_VARIABLE:
       throw new RuntimeException("Cannot translate " + expr + ". Correlated"
           + " variables should always be referenced by field access");
-    case FIELD_ACCESS:
+    case FIELD_ACCESS: {
       RexFieldAccess fieldAccess = (RexFieldAccess) expr;
+      int fieldIndex = fieldAccess.getField().getIndex();
```

Comment:
This is not required provided you don't add `"corInp" + fieldIndex` below

Reply:
The way the code is on the master branch it is needed even if "corInp" + fieldIndex is removed.

Reply:
That's fine then